website/docs/r/static_web_app_custom_domain.html.markdown

```diff
@@ -0,0 +1,115 @@
+description: |-
+  Manages a Static Web App Custom Domain.
+---
+
+# azurerm_static_web_app_custom_domain
+
+Manages a Static Web App Custom Domain.
+
+!> DNS validation polling is only done for CNAME records, terraform will not validate TXT validation records are complete.
+
+## Example Usage
+
+### CNAME validation
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_static_web_app" "example" {
+  name                = "example"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_dns_cname_record" "example" {
+  name                = "my-domain"
+  zone_name           = "contoso.com"
+  resource_group_name = azurerm_resource_group.example.name
+  ttl                 = 300
+  record              = azurerm_static_web_app.example.default_host_name
+}
+
+resource "azurerm_static_web_app_custom_domain" "example" {
+  static_web_app_id = azurerm_static_web_app.example.id
+  domain_name       = "${azurerm_dns_cname_record.example.name}.${azurerm_dns_cname_record.example.zone_name}"
+  validation_type   = "cname-delegation"
+}
+```
+
+### TXT validation
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_static_web_app" "example" {
+  name                = "example"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_static_web_app_custom_domain" "example" {
+  static_web_app_id = azurerm_static_web_app.example.id
+  domain_name       = "my-domain.contoso.com"
+  validation_type   = "dns-txt-token"
+}
+
+resource "azurerm_dns_txt_record" "example" {
+  name                = "_dnsauth.my-domain"
+  zone_name           = "contoso.com"
+  resource_group_name = azurerm_resource_group.example.name
+  ttl                 = 300
+  record {
+    value = azurerm_static_web_app_custom_domain.example.validation_token
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `domain_name` - (Required) The Domain Name which should be associated with this Static Site. Changing this forces a new Static Site Custom Domain to be created.
+
+* `static_web_app_id` - (Required) The ID of the Static Site. Changing this forces a new Static Site Custom Domain to be created.
+
+* `validation_type` - (Required) One of `cname-delegation` or `dns-txt-token`. Changing this forces a new Static Site Custom Domain to be created.
+
+-> **NOTE:** Apex domains must use `dns-txt-token` validation.
+
+-> **NOTE:** Validation using `dns-txt-token` is performed asynchronously and Terraform does wait for the validation process to be successful. Please ensure that the appropriate TXT record is created using the `validation_token` value for this to complete. 
```

Comment:
Terraform does wait or doesn't wait for the validation process to be successful?